### ISSUE TITLE
Neutralino.filesystem.readFile memory leak on Linux when file does not exist

### ISSUE THREAD

**Issue by xaxousis** [neutralinojs/neutralinojs#255]
**Expected Behavior**  
No memory leak, an error in the returned value.
  
**Actual Behavior**
The callbacks don't seem to be called, and the system memory is gobbled up.

**Steps to Reproduce the Problem**

`app.js`
```js
Neutralino.init({
    load: function() {
        Neutralino.filesystem.readFile("aaaa", v => console.log(v), console.error("error"));
    },
    pingSuccessCallback : function() {},
    pingFailCallback : function() {}
});
```

**Specifications**  
* NeutralinoJs Version: 1.4.0
* Platform: Linux Mint

Probaly related to #253.

**Comment by xaxousis** [comment]
I looked a bit around and found out that you don't check for errors when reading files...

The following function has several problems.
https://github.com/neutralinojs/neutralinojs/blob/08755204be7607fe07874247476721dba8adce91/core-linux/src/settings.cpp#L35-L46

  - It should check if the `ifstream` is correct before reading. In our case, `t.eof()` does not indicate that the file did not exist and you end in a infinite loop (which explains the memory leak since you append a `\n` at each iteration).
  - It appends a `\n` even when the file is empty.
  - You don't need to close the stream, the destructor will do it for you.

A slightly better implementation would look like the following (:warning: I haven't checked this code and am writing from memory). You'd still need to add some other checks.

```cpp
std::string getFileContent(std::string filename){
  std::ifstream t(filename, std::ios::ate);
  if(t) { // check that the file was opened
    std::ifstream::pos_type size = t.tellg();
    std::string buffer(size, '\0');
    t.seekg(0);
    t.read(buffer.data(), size);
    if(! t.fail()) {
      return buffer;
    }
  }
  // EXIT WITH AN ERROR
  return "";
}
```

**Comment by grehkil** [comment]
I have the exact same issue on Windows 10 (up to date), with Neutralino v1.3.0, using Neutralino.filesystem.readFile and Neutralino.storage.getData : when the specified file does not exists, a memory leak occurs and the error function is not called.

**Comment by oraknabo** [comment]
I see a lot of issues about this happening on cancel or on files that don't exist, but I get this behavior on linux for even files that exist. There is always a memory leak and no contents are returned.

I'm using os.dialogOpen to get the path, so I know the file is real, but it seems like the $.ajax call in filesystem.readFile fails silently without ever firing the success or error callbacks. I've tried this with both text and json content and it happens every time.

I've had no problem with filesystem.writeFile, but readFile never works for me.

**Comment by shalithasuranga** [comment]
This is fixed and will be released with v1.5.0. Please refer last comment of https://github.com/neutralinojs/neutralinojs/issues/181 for more information
